lib/AST/NameLookup.cpp

```diff
@@ -1973,10 +1974,32 @@ resolveTypeDeclsToNominal(Evaluator &evaluator,
         = evaluator(UnderlyingTypeDeclsReferencedRequest{typealias});
       auto underlyingNominalReferences
         = resolveTypeDeclsToNominal(evaluator, ctx, underlyingTypeReferences,
-                                    modulesFound);
+                                    modulesFound, anyObject);
       nominalDecls.insert(nominalDecls.end(),
                           underlyingNominalReferences.begin(),
                           underlyingNominalReferences.end());
+
+      // Recognize Swift.AnyObject directly.
```

Comment:
Swift.AnyObject should have been deserialized and type checked -- would that make this simpler?

Reply:
No, it doesn't, because we need this check to work when building the standard library, and don't want to rely on `Swift.AnyObject` having been type-checked first.